test/e2e/network/network_policy.go

```diff
@@ -353,6 +398,197 @@ var _ = SIGDescribe("NetworkPolicy", func() {
 				testCannotConnect(f, f.Namespace, clientPodName, service, 81)
 			})
 		})
+
+		ginkgo.It("should enforce updated policy [Feature:NetworkPolicy]", func() {
+			const (
+				clientAAllowedPort    = 80
+				clientANotAllowedPort = 81
+			)
+			ginkgo.By("Creating a network policy for the Service which allows traffic from pod at a port")
+			policy := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "allow-ingress",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply to server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": podServer.Name,
+						},
+					},
+					// Allow traffic only to one port.
+					Ingress: []networkingv1.NetworkPolicyIngressRule{{
+						From: []networkingv1.NetworkPolicyPeer{{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"pod-name": "client-a",
+								},
+							},
+						}},
+						Ports: []networkingv1.NetworkPolicyPort{{
+							Port: &intstr.IntOrString{IntVal: clientAAllowedPort},
+						}},
+					}},
+				},
+			}
+			policy, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policy)
+			framework.ExpectNoError(err, "Error creating Network Policy %v: %v", policy.ObjectMeta.Name, err)
+
+			testCanConnect(f, f.Namespace, "client-a", service, clientAAllowedPort)
+			err = f.WaitForPodNotFound("client-a", framework.PodDeleteTimeout)
+			if err != nil {
+				framework.ExpectNoError(err, "Expected pod to be not found.")
+			}
+			testCannotConnect(f, f.Namespace, "client-b", service, clientAAllowedPort)
+			err = f.WaitForPodNotFound("client-b", framework.PodDeleteTimeout)
+			if err != nil {
+				framework.ExpectNoError(err, "Expected pod to be not found.")
+			}
```

Comment:
The above 3 lines (line 444-446) can be merged into a single `framework.ExpectNoError(err, "Expected pod to be not found.")`